specification/network/resource-manager/Microsoft.Network/stable/2020-05-01/applicationGateway.json

```diff
@@ -2076,6 +2373,246 @@
+        },
+        "name": {
+          "type": "string",
+          "description": "Name of the private link configuration that is unique within an Application Gateway."
+        },
+        "etag": {
+          "readOnly": true,
+          "type": "string",
+          "description": "A unique read-only string that changes whenever the resource is updated."
+        },
+        "type": {
+          "readOnly": true,
+          "type": "string",
+          "description": "Type of the resource."
+        }
+      },
+      "allOf": [
+        {
+          "$ref": "./network.json#/definitions/SubResource"
+        }
+      ],
+      "description": "Private Link Configuration on an application gateway."
+    },
+    "ApplicationGatewayPrivateLinkConfigurationProperties": {
+      "properties": {
+        "ipConfigurations": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/ApplicationGatewayPrivateLinkIpConfiguration"
+          },
+          "description": "An array of application gateway private link ip configurations."
+        },
+        "provisioningState": {
+          "readOnly": true,
+          "$ref": "./network.json#/definitions/ProvisioningState",
+          "description": "The provisioning state of the application gateway private link configuration."
+        }
+      },
+      "description": "Properties of private link configuration on an application gateway."
+    },
+    "ApplicationGatewayPrivateLinkIpConfiguration": {
+      "properties": {
+        "properties": {
+          "x-ms-client-flatten": true,
+          "$ref": "#/definitions/ApplicationGatewayPrivateLinkIpConfigurationProperties",
+          "description": "Properties of an application gateway private link ip configuration."
+        },
+        "name": {
+          "type": "string",
+          "description": "The name of application gateway private link ip configuration."
+        },
+        "etag": {
+          "readOnly": true,
+          "type": "string",
+          "description": "A unique read-only string that changes whenever the resource is updated."
+        },
+        "type": {
+          "readOnly": true,
+          "type": "string",
+          "description": "The resource type."
+        }
+      },
+      "allOf": [
+        {
+          "$ref": "./network.json#/definitions/SubResource"
+        }
+      ],
+      "description": "The application gateway private link ip configuration."
+    },
+    "ApplicationGatewayPrivateLinkIpConfigurationProperties": {
+      "properties": {
+        "privateIPAddress": {
+          "type": "string",
+          "description": "The private IP address of the IP configuration."
+        },
+        "privateIPAllocationMethod": {
+          "$ref": "./network.json#/definitions/IPAllocationMethod",
+          "description": "The private IP address allocation method."
+        },
+        "subnet": {
+          "$ref": "./virtualNetwork.json#/definitions/Subnet",
```

Comment:
>Subnet [](start = 54, length = 6)

this should be the subresource definition, not a full subnet

Reply:
resolved